File: 40-50 arası örnekler/50.ortalama_kilo_hesaplama.py

girilenKilo=input("Kilonuzu giriniz: ")
girilenBoy=input("Boyunuzu giriniz: ")
girilenYontem=input('Lütfen yöntemi giriniz: ')




if girilenKilo != '' or girilenBoy != '' or girilenYontem != '':
    
    k=float(girilenKilo)
    b=float(girilenBoy)
    y=int(girilenYontem)
    
    c=''

    if y == 1 or y == 2 :
        c=str(input("Cinsiyetinizi giriniz: "))


    def yontem1(k,b,c):
        bki=0
        if c == 'e':
            bki=50+2.3*(b/2.54-60)
        else:
            bki=45.5+2.3*(b/2.54-60)
        print("ideal boy kilo oranınız: ",bki)
        
        
    yas=''

    if y == 2:
        yas=int(input("Yaşınızı giriniz: "))

    def yontem2(k,b,c,yas):
        bki=0
        if c == 'e':
            bki=(b-100+yas/10)*0.9
        else:
            bki=(b-100+yas/10)*0.8
        print("İdeal boy kilo oranınız: ",bki)
        
        
    def yontem3(k,b):
        bki=k/(b*b)
        if (bki<=17.5):
            print("Anoreksi,aşırı zayıf,yüksek risk")
        elif ((bki>17.5) and (bki<20.7)):
            print("Zayıf")
        elif ((bki>=20.7) and (bki<26.4)):
            print("Normal kilolu")
        elif ((bki>=26.4) and (bki<27.8)):
            print("Biraz fazla kilolu")
        elif ((bki>=27.8) and (bki<31.1)):
            print("Fazla kilolu")
        elif ((bki>=31.1) and (bki<34.9)):
            print("Çok fazla kilolu")
        elif ((bki>=35) and (bki<40)):
            print("Sağlık açısısndan yüksek riskli kilolu")
        elif ((bki>=40) and (bki<50)):
            print("Hastalıklı bir şekilde kilolu")
        else:
            print("Süper aşırı kilolu")
            
        
        
        
    def yontem4(b,k):
        bki=k/(b*b)
        if (bki<18.5):
            print("Zayıf")
        elif ((bki>=18.5) and (bki<25)):
            print("Orta")
        elif ((bki>=25) and (bki<30)):
            print("Fazla kilolu")
        elif ((bki>=30) and (bki<35)):
            print("1.derece obez")    
        elif ((bki>=35) and (bki<40)):
            print("2.derece obez")
        else:
            print("3.derece morbid obez")
            


    if y == 1:
        yontem1(k, b, c)
    if y == 2:
        yontem2(k, b, c, yas)
    if y == 3:
        yontem3(k, b,)
    if y == 4:
        yontem4(b, k)
else:
    print("Lütfen girdiğiniz değerleri kontrol ediniz!!!")

File: c/40-50 arası örnekler/50_ortalama_kilo_hesaplama.c
#include <stdio.h>
#include <math.h>
int main()
{
    int b, y, bki, k, yontem;
    char c;
    printf("Yontemi giriniz (1,2,3,4): ");
    scanf("%d", &yontem);

    if (yontem == 1)
    {
        printf("Boyunuzu giriniz: ");
        scanf("%d", &b);
        printf("Cinsiyetinizi giriniz : ");
        scanf("%s", &c);
        if ((c == 'E') || (c == 'e'))
            bki = 50.5 + 2.3 * (b - 60);
        else
            bki = 45.5 + 2.3 * (b - 60);
        printf("Ideal vucut agirligi : %d\n", bki);
    }

    if (yontem == 2)
    {
        printf("Boyunuzu giriniz: ");
        scanf("%d", &b);
        printf("Yasinizi giriniz: ");
        scanf("%d", &y);
        printf("Cinsiyetinizi giriniz : ");
        scanf("%s", &c);
        if ((c == 'E') || (c == 'e'))
            bki = (b - 100 + y / 10) * 0.9;
        else
            bki = (b - 100 + y / 10) * 0.8;
        printf("Ideal vucut agirligi : %d\n", bki);
    }

    if (yontem == 3)
    {
        printf("Boyunuzu giriniz: ");
        scanf("%d", &b);
        printf("Kilonuzu giriniz: ");
        scanf("%d", &k);
        bki = k / (b * b);
        if (bki <= 17.5)
            printf("Anoreksi,asiri zayif,yuksek risk");
        else if ((bki > 17.5) && (bki < 20.7))
            printf("Zayif");
        else if ((bki >= 20.7) && (bki < 26.4))
            printf("Normal kilolu");
        else if ((bki >= 26.4) && (bki < 27.8))
            printf("Biraz fazla kilolu");
        else if ((bki >= 27.8) && (bki < 31.1))
            printf("Fazla kilolu");
        else if ((bki >= 31.1) && (bki < 34.9))
            printf("Cok fazla kilolu");
        else if ((bki >= 35) && (bki < 40))
            printf("Saglık acisindan yuksek riskli kilolu");
        else if ((bki >= 40) && (bki < 50))
            printf("Hastalikli bir sekilde kilolu");
        else
            printf("Süper asiri kilolu");
    }

    if (yontem == 4)
    {
        printf("Boyunuzu giriniz: ");
        scanf("%d", &b);
        printf("Kilonuzu giriniz:\n ");
        scanf("%d", &k);
        bki = k / (b * b);
        if (bki < 18.5)
            printf("Zayif");
        else if ((bki >= 18.5) && (bki < 25))
            printf("Orta");
        else if ((bki >= 25) && (bki < 30))
            printf("Fazla kilolu");
        else if ((bki >= 30) && (bki < 35))
            printf("1.derece obez");
        else if ((bki >= 35) && (bki < 40))
            printf("2.derece obez");
        else
            printf("3.derece morbid obez");
    }
    return 0;
}